frontend/src/app/user/user-settings/user-settings.component.ts

```diff
@@ -0,0 +1,156 @@
+            .pipe(
+                filter(user => !!user),
+                takeUntilDestroyed(this.destroyRef)
+            ).subscribe(user => {
+                if (!user) {
+                    return;
+                }
+                this.form.patchValue(user);
+                this.currentUsername = user.username;
+            });
+
+        this.authService.passwordForgottenResult$
+            .pipe(
+                takeUntilDestroyed(this.destroyRef)
+            ).subscribe(result => {
+                if (!('error' in result)) {
+                    this.toastService.show({
+                        header: 'Password reset email sent',
+                        body: 'An email has been sent to you with instructions on how to reset your password.',
+                        type: 'success'
+                    });
+                }
+            });
+
+        this.authService.deleteUserResult$
+            .pipe(
+                takeUntilDestroyed(this.destroyRef)
+            ).subscribe(result => {
+                if ('error' in result) {
+                    this.toastService.show({
+                        header: 'Error deleting account',
+                        body: 'An error occurred while deleting your account. Please try again later.',
+                        type: 'danger'
+                    });
+                } else {
+                    this.toastService.show({
+                        header: 'Account deleted',
+                        body: 'Your account has been successfully deleted.',
+                        type: 'success'
+                    });
+                }
+            });
+
+        this.authService.updateSettingsResult$
+            .pipe(
+                takeUntilDestroyed(this.destroyRef)
+        ).subscribe(result => {
+            if ('error' in result) {
+                setErrors(result.error, this.form);
+            } else {
+                this.toastService.show({
+                    header: 'Settings updated',
+                    body: 'Your settings have been successfully updated.',
+                    type: 'success'
+                });
+            }
+        });
+    }
+
+    public requestPasswordReset(): void {
+        this.authService.passwordForgotten$.next({ email: this.form.getRawValue().email });
+    }
+
+    public deleteAccount(): void {
+        this.authService.deleteUser$.next();
+    }
+
+    public submit(): void {
+        this.form.markAllAsTouched();
+        updateFormValidity(this.form);
+        if (this.form.invalid) {
+            return;
+        }
+        const patchInput = this.form.getRawValue();
+
+        // If we send the username along, dj-auth-rest will assume it has changed,
+        // returning a 'username already taken' error if it hasn't.
+        // Therefore we remove the username from the input if it has not changed.
+        if (patchInput.username === this.currentUsername) {
+            delete patchInput.username;
+        }
```

Comment:
This logic belongs in the `authService`, since it's about an idiosyncracy of the request. (Or, from another perspective: any component that pushes to `authService.updateSettings$` would need the same check - so this is not a quirk of this component, but of the authService.)

Reply:
Good point. Done!